common/pango.c

```diff
@@ -8,21 +8,72 @@
 #include <string.h>
 #include "log.h"
 
+static size_t escaped_text_length(const char *text) {
+	size_t length = 0;
+	char c;
+
+	while ((c = *(text++))) {
+		switch(c) {
+		case '&':
+		case '<':
+		case '>':
+			length += 4;
+			break;
+		case '\'':
+		case '"':
+			length += 6;
+			break;
+		default:
+			length += 1;
+		}
+	}
+	return length;
+}
+
+void escape_markup_text(char *dest, const char* src) {
+	char c;
+
+	while ((c = *(src++))) {
```

Comment:
Not a fan of this, it's getting unreadable. I'd prefer to use two lines instead.

Reply:
This could actually just be an iteration over `src`. Using `c` is unnecessary.